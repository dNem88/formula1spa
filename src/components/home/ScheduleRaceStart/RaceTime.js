import React from 'react'
import styles from './raceTime.module.css'
import convertRaceTime from '../../../utils/common/converRaceTime'

function RaceTime(props) {
    const [utcTime, localTime] = convertRaceTime(props.raceStartUTC)
   
    return (
        <table className={styles['race-time-container']}>
            <tbody>
                <tr>
                    <td className={styles['left-td']}>Race</td>
                    <td className={styles['right-td']}></td>
                </tr>
                <tr>
                    <td className={styles['left-td']}>Your Local Time</td>
                    <td className={styles['right-td']}>{localTime}</td>
                </tr>
                <tr>
                    <td className={styles['left-td']}>GMT Time</td>
                    <td className={styles['right-td']}>{utcTime}</td>
                </tr>
            </tbody>
        </table>
    )
}

export default RaceTime