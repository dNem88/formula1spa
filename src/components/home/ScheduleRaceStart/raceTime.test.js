import React from 'react'
import renderer from 'react-test-renderer'
import RaceTime from './RaceTime'
import { MemoryRouter } from 'react-router-dom'
const mockProps = {
    raceStartUTC: new Date(2022,2,20,4,50).toUTCString()
}

describe('Testing RaceTime',() => {
    it('Render Correct Time', () => {
        let snapshot = renderer.create(
            <MemoryRouter>
                <RaceTime {...mockProps}/>
            </MemoryRouter>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('table')
        expect(snapshot.props.className).toBe('race-time-container')
        expect(snapshot.children[0].type).toBe('tbody')
        expect(snapshot.children[0].children.length).toBe(3)
    })
})