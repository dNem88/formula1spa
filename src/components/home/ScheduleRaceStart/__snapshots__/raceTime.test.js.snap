// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RaceTime Render Correct Time 1`] = `
<table
  className="race-time-container"
>
  <tbody>
    <tr>
      <td
        className="left-td"
      >
        Race
      </td>
      <td
        className="right-td"
      />
    </tr>
    <tr>
      <td
        className="left-td"
      >
        Your Local Time
      </td>
      <td
        className="right-td"
      >
        04:50
      </td>
    </tr>
    <tr>
      <td
        className="left-td"
      >
        GMT Time
      </td>
      <td
        className="right-td"
      >
        02:50
      </td>
    </tr>
  </tbody>
</table>
`;
